crm_lead_code/models/crm_lead.py

```diff
@@ -0,0 +1,31 @@
+##############################################################################
+# For copyright and license notices, see __manifest__.py file in root directory
+##############################################################################
+
+from odoo import api, fields, models, _
+
+
+class CrmLead(models.Model):
+    _inherit = "crm.lead"
+
+    code = fields.Char(
+        string='Lead Number', required=True, default="/", readonly=True)
```

Comment:
If it is readonly, why it is required? IMO, required is not necessary and we can remove the pre_init_hook.

Reply:
The readonly attributes only ensures that this field is automatically displayed as readonly in any view.  It is also possible to create a lead via Python code or RPC.  In this case readonly has no meaning but the required flag ensures that null can't be passed to the create method.

Reply:
I was asking this because of the default value too. but you are right, as required here prevent null value to be forced.